setup.py

```diff
@@ -111,6 +111,14 @@
             "pre-commit",
         ],
         "modin-development": [
+            f"snowflake-connector-python[pandas]{CONNECTOR_DEPENDENCY_VERSION}",
```

Comment:
instead of specifying snowflake-connector-python[pandas], let's wait till #1422 is merged and then put the `modin` requirements in a variable and use those.